### PR TITLE
As an Interviewer I should be able to see summary of a completed interview

### PR DESCRIPTION
**Pull request by OrestF**
https://trello.com/c/KbjfNJhp/52-as-an-interviewer-i-should-be-able-to-see-summary-of-a-completed-interview

code review:

code review:
- [ ] @CHILL99 
- [x] @DimaSmagulov 
- [x] @krisberry 
- [x] @natalya5 
- [x] @oleg08 
- [x] @rtershak  
- [x] @vvs79 

2nd lvl review:

- [x] @Kmalynss
- [ ] @serhiym 
- [ ] @ruba-ruba

### CODE REVIEW

File: app/controllers/interviews_controller.rb [hunk 1]
```diff
@@ -13,6 +13,7 @@ def new
```
Comment (rtershak) [inline app/controllers/interviews_controller.rb]: where the end of action `new`?

Comment (DimaSmagulov) [inline app/controllers/interviews_controller.rb]: better delete this comment

File: db/schema.rb [hunk 2]
```diff
@@ -11,7 +11,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 20151219211735) do
```
Comment (ruba-ruba) [inline db/schema.rb]: this change is not needed.

File: app/helpers/interview_helper.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+module InterviewHelper
+  def bar_color(mark)
+    color = 'danger'
+    if mark == 'green'
```
Comment (ruba-ruba) [inline app/helpers/interview_helper.rb]: poor code style, you could use `elsif` or `case` is even better

File: app/controllers/interviews_controller.rb [hunk 4]
```diff
@@ -13,6 +13,7 @@ def new
 
   def show
     @interview = Interview.find(params[:id])
+    @questions = @interview.template.questions
```
Comment (ruba-ruba) [inline app/controllers/interviews_controller.rb]: may be not relaterd to current task
but `Interview` should have relation with `Question`

Comment (OrestF) [inline app/controllers/interviews_controller.rb]: Interview belongs_to: template and template has_and_belongs_to_many :questions.
Interview have relation with Question trough Template.
Here so if I understood you correctly

Comment (ruba-ruba) [inline app/controllers/interviews_controller.rb]: take a look if it is possible to add `has_many through` relation.

**Comment by OrestF** [comment]
Made refactoring  in interview_helper.rb
Added has_many :questions relation to interview model.

**Comment by vvs79** [comment]
reviewed

File: app/controllers/interviews_controller.rb (hunk 4 above)
Comment (OrestF) [inline app/controllers/interviews_controller.rb]: I added this relation in interview model and remove 'template' from @question.template.questions, but it does't work. We became error on summary page.

**Comment by DimaSmagulov** [comment]
reviewed

File: app/helpers/interview_helper.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+module InterviewHelper
+  def bar_color(mark)
+    color = 'danger'
```
Comment (ruba-ruba) [inline app/helpers/interview_helper.rb]: this helper can be written without using `color` variable

Comment (OrestF) [inline app/helpers/interview_helper.rb]: Removed 'color' variable and fixed indents

File: app/controllers/interviews_controller.rb (hunk 4 above)
Comment (ruba-ruba) [inline app/controllers/interviews_controller.rb]: somehow it is working for me

Comment (ruba-ruba) [inline app/controllers/interviews_controller.rb]:
```Ruby
[14] pry(main)> i = Interview.find(1)
  Interview Load (0.2ms)  SELECT  `interviews`.* FROM `interviews` WHERE `interviews`.`id` = 1 LIMIT 1
=> #<Interview:0xa96b798
 id: 1,
# 
[15] pry(main)> i.template
  Template Load (0.3ms)  SELECT  `templates`.* FROM `templates` WHERE `templates`.`id` = 1 LIMIT 1
=> #<Template:0xa5b5874 id: 1, name: "Something", question_id: nil, created_at: Fri, 25 Dec 2015 15:00:00 UTC +00:00, updated_at: Fri, 25 Dec 2015 15:00:00 UTC +00:00>
[16] pry(main)> i.template.questions
  Question Load (0.4ms)  SELECT `questions`.* FROM `questions` INNER JOIN `questions_templates` ON `questions`.`id` = `questions_templates`.`question_id` WHERE `questions_templates`.`template_id` = 1
=> [#<Question:0xaac736c
  id: 48,
  content: "Explain the difference between nil and false in ruby.",
  answer: "False is a boolean datatype. Nil is not a data type.",
  topic_id: 4,
  created_at: Fri, 25 Dec 2015 15:00:00 UTC +00:00,
  updated_at: Fri, 25 Dec 2015 15:00:00 UTC +00:00,
  level: "beginner">]
[17] pry(main)> i.questions
  Question Load (0.1ms)  SELECT `questions`.* FROM `questions` INNER JOIN `questions_templates` ON `questions`.`id` = `questions_templates`.`question_id` INNER JOIN `templates` ON `questions_templates`.`template_id` = `templates`.`id` WHERE `templates`.`id` = 1
=> [#<Question:0xacba5e8
  id: 48,
  content: "Explain the difference between nil and false in ruby.",
  answer: "False is a boolean datatype. Nil is not a data type.",
  topic_id: 4,
  created_at: Fri, 25 Dec 2015 15:00:00 UTC +00:00,
  updated_at: Fri, 25 Dec 2015 15:00:00 UTC +00:00,
  level: "beginner">]

Comment (OrestF) [inline app/controllers/interviews_controller.rb]: I found the problem why it didn't work at first time and fix it. Thank to you)

**Comment by rtershak** [comment]
reviewed

**Comment by krisberry** [comment]
reviewed

**Comment by natalya5** [comment]
reviewed

**Comment by oleg08** [comment]
reviewed

File: app/models/interview.rb [hunk 6]
```diff
@@ -3,10 +3,11 @@ class Interview < ActiveRecord::Base
 
   belongs_to :template
   belongs_to :user
-  has_many :interview_questions
-
+  has_one :interview_questions
```
Comment (Kmalynss) [inline app/models/interview.rb]: why did you change ```has_many``` to ```has_one```?

Comment (OrestF) [inline app/models/interview.rb]: Sorry, I did it by accident.

**Comment by Kmalynss** [comment]
couple of "new line" issues present?
Guys how do you review?

**Comment by OrestF** [comment]
Added 'new line' at the end of each file

File: app/views/interviews/show.html.slim [hunk 7]
```diff
@@ -1,23 +1,64 @@
...
+      | Jump to 
+      span.caret
+    ul.dropdown-menu.scrollable-list
+      - @questions.count.times do |s|
+        - s += 1 
+        li
+          a href="#question#{s}"
+            |#{s}
+  - n = 1      
+  - @questions.each do |q|
```
Comment (Kmalynss) [inline app/views/interviews/show.html.slim]: @OrestF please use ```question.id``` instead of ```s``` and ```n```

File: app/views/interviews/show.html.slim [hunk 8]
```diff
@@ -1,23 +1,64 @@
...
+        = @interview.lastname
+    .col-sm-6 
+      h3
+        | Information: 
+      table.table.table-condensed
+        tr
+          td
+            | Date:          
+          td
+            = @interview.updated_at.localtime.to_s[0..15]
```
Comment (Kmalynss) [inline app/views/interviews/show.html.slim]: please use (strftime)[http://apidock.com/ruby/Date/strftime] here

**Comment by OrestF** [comment]
Used strftime
Change sequence number to question.id

**Comment by Kmalynss** [comment]
gtg
